src/Microsoft.Data/ReadOnlyDataFrameBuffer.cs

```diff
@@ -0,0 +1,94 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+using System.Text;
+
+namespace Microsoft.Data
+{
+    /// <summary>
+    /// A basic immutable store to hold values in a DataFrame column. Supports wrapping with an ArrowBuffer
+    /// </summary>
+    /// <typeparam name="T"></typeparam>
+    public class ReadOnlyDataFrameBuffer<T>
+        where T : struct
+    {
+        // TODO: Change this to Memory<T>
+
+        private ReadOnlyMemory<byte> _readOnlyMemory;
```

Comment:
should be `private readonly ReadOnlyMemory<byte> _readOnlyMemory;`.  (That's a lot of readonlys)